File: test/fixtures/dart/strange_http_method.dart
import 'package:http/http.dart' as http;

void main() async {
  final url = Uri.parse('http://localhost:28139');

  final req = http.Request('wHat', url);

  final stream = await req.send();
  final res = await http.Response.fromStream(stream);
  final status = res.statusCode;
  if (status != 200) throw Exception('http.send error: statusCode= $status');

  print(res.body);
}

File: test/fixtures/ruby/strange_http_method.rb
require 'net/http'

uri = URI('http://localhost:28139')
req = Net::HTTPGenericRequest.new('wHat', true, true, uri)

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end
